### PR TITLE
feat: add Process model and management infrastructure to Frigg Core

### PR DESCRIPTION
**Pull request by seanspeaks**
Adds generic Process model and management infrastructure to support
long-running operations like CRM syncs, data migrations, and bulk operations.
Implements repository pattern and use cases with comprehensive test coverage.

### Schema Changes
- New Process model with flexible JSON fields (context, results)
- Relations to User and Integration models
- Support for process hierarchies (parent/child)
- Indexes on userId, integrationId, type, state, name

### Fields
- Core: userId, integrationId, name, type, state
- Flexible storage: context (JSON), results (JSON)
- Hierarchy: childProcesses[], parentProcessId
- Timestamps: createdAt, updatedAt

### ProcessRepositoryInterface
- Defines contract for all Process operations
- Methods: create, findById, update, findByIntegrationAndType,
  findActiveProcesses, findByName

### Implementations
- ProcessRepositoryMongo: MongoDB/Prisma implementation
- ProcessRepositoryPostgres: PostgreSQL/Prisma implementation
- ProcessRepositoryFactory: Database-agnostic factory

### CreateProcess (5 tests)
- Creates new process records
- Sets default state (INITIALIZING)
- Validates required fields
- Handles empty context/results

### UpdateProcessState (5 tests)
- Updates process state
- Merges context updates
- Preserves existing context
- Error handling for missing processes

### UpdateProcessMetrics (8 tests)
- Updates processed/success/error counts
- Calculates duration and records/second
- Estimates completion time
- Broadcasts progress via WebSocket (optional)
- Stores last 100 errors
- Handles initial empty state

### GetProcess (4 tests)
- Retrieves process by ID
- Error handling for missing processes
- Validates repository integration

### PROCESS_MANAGEMENT_QUEUE_SPEC.md
- Detailed specification for FIFO queue enhancement
- Addresses race condition in concurrent process updates
- Architecture diagrams and cost analysis
- Implementation guide for future enhancement
- Recommended timeline and priorities

## Testing

- 22 unit tests across all use cases
- 100% test coverage on process management
- Mock repositories for isolated testing
- Validates error handling and edge cases

## Architecture Benefits

✅ Generic model supports any long-running operation
✅ Flexible JSON fields adapt to different workflows
✅ Repository pattern abstracts database implementation
✅ Use case pattern enforces business logic
✅ Comprehensive test coverage ensures reliability
✅ FIFO queue spec ready for race condition prevention

## Integration Points

- Used by BaseCRMIntegration in quo--frigg
- Supports ProcessManager service
- Enables real-time progress tracking
- Foundation for admin monitoring UI

Related: #feat/scaffold-crm-integration-base-class

### CODE REVIEW

**Comment by seanspeaks** [comment]
* **#461** <a href="https://app.graphite.dev/github/pr/friggframework/frigg/461?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#457** <a href="https://app.graphite.dev/github/pr/friggframework/frigg/457?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#456** <a href="https://app.graphite.dev/github/pr/friggframework/frigg/456?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/friggframework/frigg/456?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#455** <a href="https://app.graphite.dev/github/pr/friggframework/frigg/455?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#454** <a href="https://app.graphite.dev/github/pr/friggframework/frigg/454?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/core/integrations/use-cases/update-process-metrics.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+        // Calculate ETA if we know total
+        if (context.totalRecords > 0 && context.processedRecords > 0) {
+            const remaining = context.totalRecords - context.processedRecords;
+            if (results.aggregateData.recordsPerSecond > 0) {
+                const etaMs = (remaining / results.aggregateData.recordsPerSecond) * 1000;
+                const eta = new Date(Date.now() + etaMs);
+                context.estimatedCompletion = eta.toISOString();
+            }
+        }
```
[the remark this answers is not in the document]
Comment (d-klotz) [reply packages/core/integrations/use-cases/update-process-metrics.js:173]: I don't think processedRecords can ever be greater than totalRecords.
